java/src/main/java/org/wildfly/openssl/OpenSSLEngine.java

```diff
@@ -1282,6 +1302,16 @@ public void setSSLParameters(SSLParameters sslParameters) {
                 }
                 SSL.getInstance().setSSLVerify(ssl, value, DEFAULT_CERTIFICATE_VALIDATION_DEPTH);
             }
+
+            if(clientMode) {
+                List<SNIServerName> sniHostNames = sslParameters.getServerNames();
+                if(sniHostNames != null && !sniHostNames.isEmpty()) {
+                    for(SNIServerName serverName : sniHostNames) {
+                        SSL.getInstance().setServerNameIndication(ssl,
+                            serverName.toString());
```

Comment:
I am not sure why SSLParameters.getServerNames returns a list of SNIServerName instead of a list of SNIHostName.  It would be safer to call SNIHostName.getAsciiName() than to rely on the implementation of toString().  For example:  

if (serverName.getType() == StandardConstants.SNI_HOST_NAME) {
  SSL.getInstance().setServerNameIndication(ssl, ((SNIHostName) serverName).getAsciName());
}